common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java

```diff
@@ -183,15 +184,13 @@ protected void checkSubBlockRange(long offset, long size) {
 
   public static final void copyMemory(
       MemoryBlock src, long srcOffset, MemoryBlock dst, long dstOffset, long length) {
-    assert(length <= (src.length - src.getBaseOffset()) &&
-           length <= (dst.length - dst.getBaseOffset()));
+    assert(length <= src.length && length <= dst.length);
     Platform.copyMemory(src.getBaseObject(), src.getBaseOffset() + srcOffset,
```

Comment:
so `srcOffset` is relative, which means we should make sure `src.getBaseOffset() + srcOffset + length < src.length`

Reply:
Oh, i see.